FirebaseAuthUI/FUIAuth.m

```diff
@@ -351,16 +381,49 @@ - (void)signInWithEmailHint:(NSString *)emailHint
                                                    FIRAuthResultCallback  _Nullable result) {
             if (error) {
               if (completion) {
-                completion(nil, error);
+                completion(nil, error, nil);
               }
               return;
             }
-            [tempAuth signInAndRetrieveDataWithCredential:credential completion:completion];
+
+            [tempAuth signInAndRetrieveDataWithCredential:credential
+                                               completion:^(FIRAuthDataResult *_Nullable authResult,
+                                                            NSError *_Nullable error) {
+              if (error) {
+                if (completion) {
+                  completion(nil, error, nil);
+                }
+              }
+
+              // Handle potential email mismatch.
+              if (![emailHint isEqualToString:authResult.user.email]) {
+                NSString *signedInEmail = authResult.user.email;
+                NSString *title =
+                    [NSString stringWithFormat:@"Continue sign in with %@?", signedInEmail];
+                NSString *message =
+                    [NSString stringWithFormat:@"You originally wanted to sign in with %@",
+                    emailHint];
```

Comment:
If these strings are user-facing, they should be localized.

Reply:
The entire flow needs to be localized in a follow up CL.